### ISSUE TITLE
Change docker backend to dlite

### ISSUE THREAD

**Issue by onnimonni** [IFTTT/dash#27]
I have been testing: https://github.com/nlf/dlite for few days.
**TL;DR: It is awesome.**

**dlite:**
- automatically starts in system restart
- doesn't need any eval $(dev machine env) since it mounts `docker.sock` in host machine
- ip address doesn't change.

**Comment by silvamerica** [comment]
Yeah I've been using dlite off and on for a few weeks now. I'm not sure how we can change the backend without causing a lot of issues for existing users, but I'm totally open to the possibilities!

**Comment by onnimonni** [comment]
Existing users already have issues which this would solve ;).

I think we could check existance of 'docker-machine status dev' and give a huge prompt to users about breaking changes.

I'd really rather contribute to this project instead of having my own fork with all changes. In few weeks I will start rolling new dev tools internally and dlite is a must.

Do You have Any ideas of how we could make this happen?

**Comment by silvamerica** [comment]
A few options include:
- As you said, checking for the existence of the VM already
- Adding a check in `dev update` that would prevent updating past a certain point
- Maintaining a separate branch for dlite
- Generalizing everything so that it can work with multiple backends, including docker-machine with virtualbox, docker-machine-driver-xhyve, and dlite.
...and more! :-D

All of these have tradeoffs, and aren't necessarily mutually exclusive.
